lib/virtio/virtqueue.c

```diff
@@ -28,6 +29,14 @@ static int virtqueue_nused(struct virtqueue *vq);
 static int virtqueue_navail(struct virtqueue *vq);
 #endif
 
+#ifdef VIRTIO_CACHED_VRINGS
+#define VRING_FLUSH(x)		metal_cache_flush(&x, sizeof(x))
+#define VRING_INVALIDATE(x)	metal_cache_invalidate(&x, sizeof(x))
+#else
+#define VRING_FLUSH(x)		do { } while (0)
+#define VRING_INVALIDATE(x)	do { } while (0)
```

Comment:
Thanks for the update!
I wondered if the  "do { } while (0)"  add useless assembler code, but seems not the case with GCC, and seems better to implement it in this way to avoid warnings with some old compilers.  I will cross check the point for ARMCC and EWARM compiler for the next release and fix it if needed. 
If no more remarks we will probably merge it next week

Reply:
thanks!